python-checks/src/main/java/org/sonar/python/checks/tests/NotDiscoverableTestMethodCheck.java

```diff
@@ -43,23 +47,34 @@
 public class NotDiscoverableTestMethodCheck extends PythonSubscriptionCheck {
 
   private static final String MESSAGE = "Rename this method so that it starts with \"test\" or remove this unused helper.";
+  private static final Set<String> globalFixture = new HashSet<>();
 
   @Override
   public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.FUNCDEF, NotDiscoverableTestMethodCheck::lookForGlobalFixture);
+
     context.registerSyntaxNodeConsumer(Tree.Kind.CLASSDEF, ctx -> {
       ClassDef classDefinition = (ClassDef) ctx.syntaxNode();
 
       if (inheritsOnlyFromUnitTest(classDefinition)) {
         Map<FunctionSymbol, FunctionDef> suspiciousFunctionsAndDefinitions = new HashMap<>();
         Set<Tree> allDefinitions = new HashSet<>();
+        // build set of fixtures
+        Set<String> currentClassFixture = classDefinition.body().statements().stream()
+          .filter(statement -> statement.is(Tree.Kind.FUNCDEF))
+          .map(FunctionDef.class::cast)
+          .filter(functionDef -> functionDef.decorators().stream().anyMatch(NotDiscoverableTestMethodCheck::isPytestFixture))
+          .map(functionDef -> functionDef.name().name())
+          .collect(Collectors.toSet());
+
         // We only consider method definitions, and not nested functions
         for (Statement statement : classDefinition.body().statements()) {
           if (statement.is(Tree.Kind.FUNCDEF)) {
             FunctionDef functionDef = ((FunctionDef) statement);
             String functionName = functionDef.name().name();
             Symbol symbol = functionDef.name().symbol();
-            // If it doesn't override existing methods and doesn't start with test, it is added to the map
-            if (!overrideExistingMethod(functionName) && !functionName.startsWith("test")) {
+            // If it doesn't override existing methods, doesn't start with test and is not a helper, it is added to the map
+            if (!overrideExistingMethod(functionName) && !functionName.startsWith("test") && !isHelper(functionDef, currentClassFixture)) {
```

Comment:
Maybe combine these conditions in a simple method like boolean isException(FunctionDef functionDef) like:
```
String functionName = functionDef.name().name()
return overrideExistingMethod(functionName) || functionName.startsWith("test") || isHelper(functionDef, currentClassFixture))
```